### PR TITLE
fix(subsonic): avoid double brackets when appending subtitle or version

### PR DESCRIPTION
**Pull request by deluan**
### Description

When `Subsonic.AppendSubtitle` or `Subsonic.AppendAlbumVersion` is enabled, the subtitle/version tag was always wrapped in parentheses before being appended, so a tag that already came bracketed got doubled — e.g. `(non-explicit version)` rendered as `The Lazy Generation ((non-explicit version))`.

Now the tag is appended as-is when it is already wrapped in a matching bracket pair — `()`, `[]`, `{}` or `<>` — with surrounding whitespace trimmed first. The shared `appendSuffix` helper is used by `MediaFile.FullTitle`, `MediaFile.FullAlbumName` and `Album.FullName`, so all consumers stay consistent.

Note: the search full-text column stores the appended title at scan time, so existing rows keep their old value until the next scan; display output is corrected immediately.

### Type of Change

- [x] Bug fix
- [ ] New feature
- [ ] Documentation update
- [ ] Refactor
- [ ] Other (please describe):

### Checklist

- [x] My code follows the project's coding style
- [x] I have tested the changes locally
- [ ] I have added or updated documentation as needed
- [x] I have added tests that prove my fix/feature works (or explain why not)
- [x] All existing and new tests pass

### How to Test

1. Enable `Subsonic.AppendSubtitle` (and/or `Subsonic.AppendAlbumVersion`).
2. On a track whose `subtitle` tag is already bracketed (e.g. `(non-explicit version)`), confirm the title reads `Title (non-explicit version)` — not doubled.
3. Verify a plain subtitle like `Live` still renders as `Title (Live)`, a padded ` (Live) ` as `Title (Live)`, and a mismatched `[Live)` as `Title ([Live))`.

### CODE REVIEW

File: model/mediafile.go [hunk 1]
```diff
@@ -100,18 +100,28 @@ type MediaFile struct {
...
 
+var bracketPairs = map[byte]byte{'(': ')', '[': ']', '{': '}', '<': '>'}
+
+func appendSuffix(base, suffix string) string {
+	suffix = strings.TrimSpace(suffix)
+	if len(suffix) >= 2 && bracketPairs[suffix[0]] == suffix[len(suffix)-1] {
+		return base + " " + suffix
+	}
+	return base + " (" + suffix + ")"
+}
```
[the remark this answers is not in the document]
Comment (deluan) [reply model/mediafile.go:123]: Fixed in 4f6e8156 — appendSuffix now returns the base name unchanged when the tag is empty after trimming. Kept the package-level bracketPairs map rather than inlining byte comparisons; a 4-entry map built once is more readable and the lookup cost is negligible.

[the remark this answers is not in the document]
Comment (deluan) [reply model/mediafile.go:123]: Fixed in 4f6e8156 — returns base early when suffix is empty after TrimSpace, with a whitespace-only test case added.

[the remark this answers is not in the document]
Comment (deluan) [reply model/mediafile.go:123]: Reverted this — the whitespace-only guard was dropped. That input isn't reachable: subtitle/albumversion have no split regex in mappings.yaml, and the gotaglib extractor passes values through untouched, so there's no path that stores a spaces-only value to guard against. The branch was rebased, so commit 4f6e8156 no longer exists.

[the remark this answers is not in the document]
Comment (deluan) [reply model/mediafile.go:123]: Reverted this — the whitespace-only guard was dropped. That input isn't reachable: subtitle/albumversion have no split regex in mappings.yaml, and the gotaglib extractor passes values through untouched, so there's no path that stores a spaces-only value to guard against. The branch was rebased, so commit 4f6e8156 no longer exists.

File: model/mediafile_test.go [hunk 2]
```diff
@@ -533,6 +533,13 @@ var _ = Describe("MediaFile", func() {
...
 		Entry("returns just title when tag is absent", true, Tags{}, "Song"),
 		Entry("returns just title when tag is an empty slice", true, Tags{TagSubtitle: []string{}}, "Song"),
+		Entry("does not double parentheses when subtitle is already parenthesized", true, Tags{TagSubtitle: []string{"(non-explicit version)"}}, "Song (non-explicit version)"),
+		Entry("does not add parentheses when subtitle is wrapped in square brackets", true, Tags{TagSubtitle: []string{"[Live]"}}, "Song [Live]"),
+		Entry("does not add parentheses when subtitle is wrapped in curly braces", true, Tags{TagSubtitle: []string{"{Remix}"}}, "Song {Remix}"),
+		Entry("does not add parentheses when subtitle is wrapped in angle brackets", true, Tags{TagSubtitle: []string{"<Live>"}}, "Song <Live>"),
+		Entry("adds parentheses when brackets do not match", true, Tags{TagSubtitle: []string{"[Live)"}}, "Song ([Live))"),
+		Entry("trims surrounding whitespace before wrapping", true, Tags{TagSubtitle: []string{"  Live  "}}, "Song (Live)"),
+		Entry("trims whitespace around an already-bracketed subtitle", true, Tags{TagSubtitle: []string{" (Live) "}}, "Song (Live)"),
```
[the remark this answers is not in the document]
Comment (deluan) [reply model/mediafile_test.go:542]: Not adding this test — the whitespace-only case it covers isn't reachable (subtitle/albumversion have no split regex and the extractor doesn't trim), so the guard and its test were dropped rather than added.
